superset-frontend/packages/superset-ui-chart-controls/src/sections/matrixify.tsx

```diff
@@ -0,0 +1,159 @@
+ * under the License.
+ */
+import { t } from '@superset-ui/core';
+import { ControlPanelSectionConfig } from '../types';
+
+export const matrixifyEnableSection: ControlPanelSectionConfig = {
+  label: t('Enable Matrixify'),
+  expanded: true,
+  controlSetRows: [
+    [
+      {
+        name: 'matrixify_enabled',
+        config: {
+          type: 'CheckboxControl',
+          label: t('Enable Matrixify'),
+          default: false,
+          renderTrigger: true,
+          description: t(
+            'Transform this chart into a matrix/grid of charts based on dimensions or metrics',
+          ),
+        },
+      },
+    ],
+  ],
+  tabOverride: 'matrixify',
+};
+
+export const matrixifySection: ControlPanelSectionConfig = {
+  label: t('Matrixify'),
+  expanded: false,
+  visibility: ({ controls }) => controls?.matrixify_enabled?.value === true,
+  controlSetRows: [
+    [
+      {
+        name: 'matrixify_show_row_labels',
+        config: {
+          type: 'CheckboxControl',
+          label: t('Show row labels'),
+          default: true,
+          renderTrigger: true,
+          description: t('Whether to show row labels in the matrix grid'),
+        },
+      },
+      {
+        name: 'matrixify_show_column_headers',
+        config: {
+          type: 'CheckboxControl',
+          label: t('Show column headers'),
+          default: true,
+          renderTrigger: true,
+          description: t('Whether to show column headers in the matrix grid'),
+        },
+      },
+    ],
+    [
+      {
+        name: 'matrixify_row_height',
+        config: {
+          type: 'TextControl',
+          label: t('Row height'),
+          default: 300,
+          isInt: true,
+          renderTrigger: true,
+          description: t('Height of each row in pixels'),
+        },
+      },
+      {
+        name: 'matrixify_fit_columns_dynamically',
+        config: {
+          type: 'CheckboxControl',
+          label: t('Fit columns dynamically'),
+          default: true,
+          renderTrigger: true,
+          description: t('Automatically adjust column width based on content'),
+        },
+      },
+    ],
+    [
+      {
+        name: 'matrixify_charts_per_row',
+        config: {
+          type: 'TextControl',
+          label: t('Charts per row'),
+          default: 3,
+          isInt: true,
+          renderTrigger: true,
+          description: t(
+            'Number of charts per row when not fitting dynamically',
+          ),
+          visibility: ({ controls }) =>
+            !controls?.matrixify_fit_columns_dynamically?.value,
+        },
+      },
+    ],
+    [
+      {
+        name: 'matrixify_cell_title_template',
+        config: {
+          type: 'TextControl',
+          label: t('Cell title template'),
+          default: '',
+          description: t(
+            'Template for cell titles. Use Handlebars syntax: {{row}}, {{column}}, {{rowLabel}}, {{columnLabel}}',
```

Comment:
I suspect that many non-technical users may not know what  Handlebars is. How about “double curly braces” or “brackets”?